cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/DefaultSecureStoreService.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.services;
+
+import co.cask.cdap.api.security.store.SecureStore;
+import co.cask.cdap.api.security.store.SecureStoreData;
+import co.cask.cdap.api.security.store.SecureStoreManager;
+import co.cask.cdap.api.security.store.SecureStoreMetadata;
+import co.cask.cdap.common.BadRequestException;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.proto.id.SecureKeyId;
+import co.cask.cdap.proto.security.Action;
+import co.cask.cdap.proto.security.Principal;
+import co.cask.cdap.proto.security.SecureKeyCreateRequest;
+import co.cask.cdap.proto.security.SecureKeyListEntry;
+import co.cask.cdap.security.authorization.AuthorizerInstantiator;
+import co.cask.cdap.security.spi.authentication.SecurityRequestContext;
+import com.google.common.base.Strings;
+import com.google.inject.Inject;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Default implementation of the service that manages access to the Secure Store,
+ */
+public class DefaultSecureStoreService implements SecureStoreService {
+  private final AuthorizerInstantiator authorizerInstantiator;
+  private final SecureStore secureStore;
+  private final SecureStoreManager secureStoreManager;
+
+  @Inject
+  DefaultSecureStoreService(AuthorizerInstantiator authorizerInstantiator, SecureStore secureStore,
+                            SecureStoreManager secureStoreManager) {
+    this.authorizerInstantiator = authorizerInstantiator;
+    this.secureStore = secureStore;
+    this.secureStoreManager = secureStoreManager;
+  }
+
+  /**
+   * Lists all the secure keys in the given namespace that the user has access to.
+   * @param namespaceId Id of the namespace we want the key list for.
+   * @return A list of {@link SecureKeyListEntry} of all the keys visible to the user under the given namespace.
+   * @throws Exception If either the caller does not have READ access to the namespace or if there was a problem
+   * getting the list from the underlying provider.
+   */
+  @Override
+  public List<SecureKeyListEntry> list(NamespaceId namespaceId) throws Exception {
+    Principal principal = SecurityRequestContext.toPrincipal();
+    authorizerInstantiator.get().enforce(namespaceId, principal, Action.READ);
```

Comment:
For enforcement, you should use an ``AuthorizationEnforcer`` instead. You can inject it in the constructor as well. Also, list should not enforce. It should filter the keys to only return keys that the principal has *some* privileges on. You can take a look at #6090 for how to do that. That will be merged soon. ``AuthorizationEnforcer`` is already in develop, just the signature of the ``filter`` method changes  in #6090